cluster-autoscaler/core/dynamic_autoscaler.go

```diff
@@ -0,0 +1,100 @@
+package core
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/contrib/cluster-autoscaler/config/dynamic"
+	"k8s.io/contrib/cluster-autoscaler/metrics"
+
+	kube_client "k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
+	kube_record "k8s.io/kubernetes/pkg/client/record"
+
+	"github.com/golang/glog"
+)
+
+// DynamicAutoscaler is a variant of autoscaler which supports dynamic reconfiguration at runtime
+type DynamicAutoscaler struct {
+	autoscaler Autoscaler
+	autoscalerBuilder    AutoscalerBuilder
+	configFetcher dynamic.ConfigFetcher
+}
+
+// NewDynamicAutoscaler builds a DynamicAutoscaler from required parameters
+func NewDynamicAutoscaler(autoscalerBuilder AutoscalerBuilder, configFetcher dynamic.ConfigFetcher) *DynamicAutoscaler {
```

Comment:
I began to think that we can completely hide `DynamicAutoscaler`, `NewDynamicAutoscaler`, `StaticAutoscaler` and `NewStaticAutoscaler` from other packages by making these private i.e. `dynamicAutoscaler` and `newDynamicAutoscaler` respectively, so that we can ensure that `NewAutoscaler` is used in `main.go`.